Source/AviationWeather/Source/metar.cpp

```diff
@@ -1395,6 +1400,32 @@ void metar_info::parse()
     detail::parse_altimeter(*this, baseMetar);
 }
 
+cloud_layer metar_info::ceiling() const
+{
+    for (auto it = sky_condition_group.begin(); it != sky_condition_group.end(); ++it)
+    {
+        if (it->sky_cover == sky_cover_type::broken || it->sky_cover == sky_cover_type::overcast)
+        {
+            return *it;
+        }
+    }
+    // check vertical visibility?
+    return cloud_layer();
+}
+
+flight_category metar_info::flight_category() const
+{
+    auto ceiling = metar_info::ceiling();
+    if (visibility_group >= visibility(3.0, distance_unit::statute_miles) && ceiling.layer_height >= 1000L)
```

Comment:
use aw::convert(visibility_group::distance, visibility_group::unit, distance_unit::statute_miles) >= 3.0 instead of constructing a new visibility object. Same applies below.